react/javascript/test/HighlightTest.tsx

```diff
@@ -74,74 +77,18 @@ describe('HighLight', () => {
     ])
   })
 
-  context('when htmlContent is not set', () => {
-    it('escapes HTML characters', () => {
```

Comment:
We now won't have a test that proves the XSS protection, is that okay?

Reply:
Good call - I'll add one

Reply:
I've added a couple more tests for XSS. It strips away `<script>` tags and allows all other HTML tags.

I'm not entirely sure if we should be that permissive though - it sort of defeats the purpose of Markdown to allow any HTML. Should we be more restrictive?

Reply:
Good question. Most markdown implementations seem to allow HTML by default, but I think in a Gherkin document I can't see a use case for doing that. The only thing where Markdown semantics fail a bit

Reply:
Good question. Most markdown implementations seem to allow HTML by default, but in context of a Gherkin document I can't see a use case for doing that. I would say start out restrictive and if there's demand for broad HTML support we can look at why.

(I write a lot of Markdown, the only HTML tag I can recall writing recently is <small>, other than that I think the syntax is more than rich enough.)

Reply:
The standalone HTML report has a search function which will highlight searches with `<mark>`, so we could allow `<small>` and `<mark> for now.

But that seems a bit arbitrary. Are there any downsides to allowing anything but `<script>`?

Reply:
In my opinion, the main issue is not the HTML tags themselves but the possibility to have Javascript executed with `onclick` events etc.
Is there a way with `react-markdown` to ensure there's no Javascript added via HTML tag attributes ? If so, we could allow any HTML tag* without having to question too much the safety :)

* well, any tag except `<script>` :D

Reply:
Good point @vincent-psarga! More tests to add then....